test/odbc/psqlodbc/sql_variant.cpp

```diff
@@ -0,0 +1,614 @@
+  vector<string> &casted_inserted_values, vector<string> &bbf_expected, vector<string> &pg_expected) {
+  // Clear all values for output
+  casted_inserted_values.clear();
+  bbf_expected.clear();
+  pg_expected.clear();
+  
+  // Get inserted_values
+  for (int i = 0; i < INSERTION_INFO.size(); i++) {
+    string insert_value     =  std::get<0>(INSERTION_INFO[i]);
+    string pg_expected_val  =  std::get<1>(INSERTION_INFO[i]);
+    string type             =  std::get<2>(INSERTION_INFO[i]);
+    bool needs_quote        =  std::get<3>(INSERTION_INFO[i]);
+    bool is_binary          =  type.find("binary") != std::string::npos;
+    string insert_string    =  "";
+
+    if (insert_value.compare("NULL") == 0) {
+      insert_string = insert_value;
+    }
+    else if (needs_quote) {
+      insert_string = "CAST('" + insert_value + "' AS " + type + ")";
+    }
+    else if (is_binary) {
+      insert_string = "CAST(" + HexToIntStr(insert_value) + " AS " + type + ")";
+    }
+    else {
+      insert_string = "CAST(" + insert_value + " AS " + type + ")";
+    }
+
+    casted_inserted_values.push_back(insert_string);
+    bbf_expected.push_back(insert_value);
+    pg_expected.push_back(pg_expected_val.empty() ? insert_value : pg_expected_val);
+  }
+  return;
+}
+
+// Helper function for SqlVariant_InitOperationsQuery to construct queries
+void SqlVariant_OperationsQueryBuilder(const int &pk1, const int &pk2, const string &comparison, 
+  vector<string> &bbf_operations_query, vector<string> &pg_operations_query) {
+  string bbf_query = "SELECT IIF((" + SelectStatement(BBF_TABLE_NAME, {COL2_NAME}, {}, COL1_NAME + "=" + std::to_string(pk1))
+        + ")" + comparison + "("  + SelectStatement(BBF_TABLE_NAME, {COL2_NAME}, {}, COL1_NAME + "=" + std::to_string(pk2))
+        + "), '1', '0')";
+
+  string pg_query = "SELECT (" + SelectStatement(PG_TABLE_NAME, {COL2_NAME}, {}, COL1_NAME + "=" + std::to_string(pk1))
+    + ") operator(sys." + comparison + ") ("  + SelectStatement(PG_TABLE_NAME, {COL2_NAME}, {}, COL1_NAME + "=" + std::to_string(pk2))
+    + ")";
+
+  bbf_operations_query.push_back(bbf_query);
+  pg_operations_query.push_back(pg_query);
+  return;
+}
+
+// Helper function for PSQL_DataTypes_Sql_Variant.Comparison_Operators to create comparison operation queries 
+//  that are expected to be true
+void SqlVariant_InitOperationsQuery_True(const int &size, vector<string> &bbf_operations_query, 
+  vector<string> &pg_operations_query) {
+  bbf_operations_query.clear();
+  pg_operations_query.clear();
+
+  for (int i = 0; i < size; i++) {
+    if (i < size -1)
+    {
+      // > operator
+      SqlVariant_OperationsQueryBuilder(i, i+1, ">", bbf_operations_query, pg_operations_query);
+      // >= operator
+      SqlVariant_OperationsQueryBuilder(i, i+1, ">=", bbf_operations_query, pg_operations_query);
+      // < operator
+      SqlVariant_OperationsQueryBuilder(i+1, i, "<", bbf_operations_query, pg_operations_query);
```

Comment:
```suggestion
      SqlVariant_OperationsQueryBuilder(i + 1, i, "<", bbf_operations_query, pg_operations_query);
```
Similarly for all other relevant places.

Reply:
Sure, done